lib/eval_on_dart_library.dart

```diff
@@ -21,37 +19,33 @@ class EvalOnDartLibrary {
     });
   }
 
-  String _libraryName;
+  final String libraryName;
+  final VmService service;
   Completer<LibraryRef> _libraryRef;
-  VmService _service;
   String _isolateId;
 
   void _initialize(String isolateId) async {
     _isolateId = isolateId;
 
-    _service.getIsolate(_isolateId).then<dynamic>((dynamic value) {
-      if (value is Isolate) {
-        for (LibraryRef library in value.libraries) {
-          if (library.uri == _libraryName) {
-            _libraryRef.complete(library);
-            return;
-          }
+    service.getIsolate(_isolateId).then<dynamic>((dynamic value) {
+      for (LibraryRef library in value.libraries) {
+        if (library.uri == libraryName) {
+          _libraryRef.complete(library);
+          return;
         }
-        _libraryRef.completeError('Library $_libraryName not found.');
       }
+      _libraryRef.completeError('Library $libraryName not found.');
     }).catchError((RPCError e) => print('RPCError ${e.code}: ${e.details}'));
   }
 
-  Completer<InstanceRef> eval(String expression) {
-    final Completer<InstanceRef> future = new Completer<InstanceRef>();
-    _libraryRef.future.then((LibraryRef ref) {
-      _service
-          .evaluate(_isolateId, ref.id, expression)
-          .then((dynamic response) => future.complete(response))
-          .catchError((RPCError e) => print('RPCError ${e.code}: ${e.details}'))
-          .catchError((Error e) => print('${e.kind}: ${e.message}'))
-          .catchError((dynamic e) => print('Unrecognized error: $e'));
-    });
-    return future;
+  Future<InstanceRef> eval(String expression) async {
+    final LibraryRef libraryRef = await _libraryRef.future;
+    final InstanceRef instanceRef = await service
+        .evaluate(_isolateId, libraryRef.id, expression)
+        .then<InstanceRef>((dynamic response) => response)
```

Comment:
remove use of .then and use await syntax instead.

Reply:
Done.